src/xunit.assert/Asserts/Sdk/AssertEqualityComparer.cs

```diff
@@ -149,6 +158,36 @@ public bool Equals(T x, T y)
             return dictionaryYKeys.Count == 0;
         }
 
+        bool? CheckIfSetsAreEqual(T x, T y, TypeInfo typeInfo)
+        {
+            if (!IsSet(typeInfo))
+                return null;
+
+            var enumX = x as IEnumerable;
+            var enumY = y as IEnumerable;
+            if (enumX == null || enumY == null)
+                return null;
+
+            var objectEnumX = enumX.Cast<object>();
+            var objectEnumY = enumY.Cast<object>();
+            if (objectEnumX.Count() != objectEnumY.Count())
+                return false;
+
+            var setX = new HashSet<object>(objectEnumX, objectComparer);
```

Comment:
No, I _do actually mean_ the T in `ISet<T>` here (not the T in `Assert.Equal<T>`).